### PR TITLE
Prims Algorithm implementation in Dart

### LINKED ISSUE #2201: Prims Algorithm
- [x] C
- [x] C++
- [ ] C# - @ritish099
- [ ] CoffeeScript
- [ ] Dart - @Swarnimashukla
- [ ] Go - @adi44
- [x] Java
- [x] JS - @sukritishah15
- [x] Kotlin - @darkmatter18
- [ ] Php - @prince-09
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript - @BhuwanChandra

### PR DESCRIPTION
**Pull request by Swarnimashukla**
Fixes #2201 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Prims Algorithm implementation in Dart

#### Languages Used:
- Dart

#### Files Added:
- Prims_Algorithm.dart

@jainaman224 @asha15 @raghav-dalmia @SakshiShreya

### CODE REVIEW

**Comment by Swarnimashukla** [comment]
@jainaman224   @asha15  please review this PR .

File: Prims_Algorithm/Prims_Algorithm.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+uses the greedy approach. Prim's algorithm shares a similarity with the
+shortest path first algorithms.
+
+Prim's algorithm, in contrast with Kruskal's algorithm, treats the nodes as a
+single tree and keeps on adding new nodes to the spanning tree from the
+given graph.
+ */
+
+import 'dart:io';
+
```
Comment (somya-kapoor) [inline Prims_Algorithm/Prims_Algorithm.dart:12]: remove this line

File: Prims_Algorithm/Prims_Algorithm.dart [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  \        |        /
+   \       |       /
+    \      2      /
+     6     |     8
+      \    |    /
+       \   |   /
+        \  |  /
+         \ | /
+          (4)
+
```
Comment (somya-kapoor) [inline Prims_Algorithm/Prims_Algorithm.dart:149]: remove this line

File: Prims_Algorithm/Prims_Algorithm.dart [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
```
Comment (somya-kapoor) [inline Prims_Algorithm/Prims_Algorithm.dart:150]: remove this line

File: Prims_Algorithm/Prims_Algorithm.dart [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    else
+    {
+      adjmat[src][dest] = weight;
+    }
+  }
+
+  prim_mst( adjmat, n + 1 );
+
```
Comment (somya-kapoor) [inline Prims_Algorithm/Prims_Algorithm.dart:118]: remove this line

File: Prims_Algorithm/Prims_Algorithm.dart [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+void printMST( parent, graph, V )
+{
+  print("Edge \tWeight\n");
+  for ( var i = 1; i < V; i++ ) {
+    print('${parent[i]} - ${i} \t ${graph[i][parent[i]]} \n');
+  }
+}
+
+
```
Comment (somya-kapoor) [inline Prims_Algorithm/Prims_Algorithm.dart:41]: remove this line

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by Swarnimashukla** [comment]
Yes I will submit my code by Tomorrow morning

**Comment by Swarnimashukla** [comment]
@somya-kapoor  I made recommended changes please go through it .

**Pull request merged**